inc/RestApi/PluginsController.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Bluehost\Maestro\RestApi;
+
+use Exception;
+use WP_REST_Server;
+use WP_REST_Response;
+
+use Bluehost\Maestro\Plugin;
+use Bluehost\Maestro\WebPro;
+
+/**
+ * Class PluginsController
+ */
+class PluginsController extends \WP_REST_Controller {
+
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'bluehost/maestro/v1';
+
+	/**
+	 * The current Web Pro accessing the endpoint
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var WebPro
+	 */
+	private $webpro;
+
+	/**
+	 * Registers the Plugins routes
+	 *
+	 * @since 1.1.1
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/plugins',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_plugins' ),
+					'permission_callback' => array( $this, 'check_permission' ),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Callback for the plugins get endpoint
+	 *
+	 * Returns a list of installed plugins with details and updates
+	 *
+	 * @since 1.1.1
+	 *
+	 * @return WP_Rest_Response Returns a standard rest response with a list of plugins
+	 */
+	public function get_plugins() {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			include_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		if ( ! function_exists( 'get_option' ) ) {
+			include_once ABSPATH . 'wp-includes/options.php';
+		}
+
+		// Make sure we populate the plugins updates transient
+		wp_update_plugins();
+
+		$installed_plugins = get_plugins();
+		$updates           = get_site_transient( 'update_plugins' );
+		$auto_updates      = get_option( 'auto_update_plugins' );
+		$plugins           = array();
+
+		foreach ( $installed_plugins as $plugin_slug => $plugin_details ) {
+			$plugin = new Plugin( $plugin_slug, $updates, $plugin_details, $auto_updates );
+			array_push( $plugins, $plugin );
+		}
+
+		return new WP_Rest_Response(
+			array(
+				'plugins'            => $plugins,
+				'auto_update_global' => get_option( 'auto_update_plugin' ),
```

Comment:
This option is specific to the Bluehost plugin and will only be present if it has been installed and active at a point in the past. If the plugin is accurate, then this is likely to be inaccurate.

There is no global "on/off" switch for plugins and themes in WordPress Core. There are a few filters that allow plugins and themes to control this, but I don't recommend checking against the values they return as they are deep within the internals of the update API.

A bit about how the Bluehost plugin controls plugin and theme auto-update settings. There is an on/off switch for each in the plugin. But when those are toggled, the plugin takes every active plugin|theme and inserts them into the default WordPress Core `auto_update_plugins|auto_update_themes` options. This way, if someone disables the plugin, every plugin and theme installed at that point will continue to auto-update.

So when we are looking at whether a plugin or theme will auto-update, the recommendation is to use `auto_update_plugins|auto_update_themes` and check for the presence of the plugin's slug.

Reply:
So we can selectively get this option if the bluehost plugin is installed / include that functionality with this one as well, will need to get back to you on this after discussing it with my team.

Reply:
I wouldn't use this option at all. The Bluehost plugin syncs all update options to the default Core options. Those will always have the correct auto-update information whether the plugin is active (now or previously), or not.

Reply:
I have used this option but have a check before that ensures that the BH plugin is installed and active, that should work fine then I think because that is the methods used by our servers to update these settings